gunagala/imager.py

```diff
@@ -1561,18 +1577,19 @@ def make_noiseless_image(self,
         """
         electrons = np.zeros((self.wcs._naxis2,
                               self.wcs._naxis1)) * u.electron / (u.second * u.pixel)
+        self.set_WCS_centre(centre)
 
         # Calculate observed sky background
         sky_rate = self.sky_rate[filter_name]
         if hasattr(self.sky, 'relative_brightness'):
-            pixel_coords = self.get_pixel_coords(centre)
+            pixel_coords = self.get_pixel_coords()
             relative_sky = self.sky.relative_brightness(pixel_coords, obs_time)
             sky_rate = sky_rate * relative_sky
         electrons = electrons + sky_rate
 
         if stars is not None:
             for (coords, magnitude) in stars:
-                coords = SkyCoord(coords)
+                coords = SkyCoord(coords, unit='deg')
```

Comment:
As per comment at line 301, explicitly forcing units to degrees here reduces flexibility and is usually unnecessary. If you really want to be able to explicitly specify units make it an optional argument of the method so that the user can choose units, or choose not to specify them.